### PR TITLE
*: Three preparation work for `CREATE DATABASE` in declarative schema changer

### PR DESCRIPTION
**Pull request by Xiang-Gu**
This PR separates out a few less controversial and easy-to-review commits for the overall bigger effort to implement `CREATE DATABASE` in declarative schema changer. See each commit for details.

Informs #103607

### CODE REVIEW

File: pkg/sql/schemachanger/scbuild/builder_state.go [hunk 1]
```diff
@@ -330,6 +330,9 @@ func (b *builderState) CurrentUserHasAdminOrIsMemberOf(role username.SQLUsername
 	if b.hasAdmin {
 		return true
 	}
+	if b.evalCtx.SessionData().User() == role {
```
Comment (annrpom) [inline pkg/sql/schemachanger/scbuild/builder_state.go:333]: oh interesting - nice catch

**Review by annrpom — APPROVED** [review]
lgtm!

supy tiny nit: you have a typo in the third commit (:s/Coded/Codec) 

> This commit introduces the Coded()

**Comment by annrpom** [comment]
also it looks like this import needs to be changed as well
```
 lint_test.go:1898:
        pkg/cli/democluster/demo_cluster_test.go:30:2: "github.com/cockroachdb/cockroach/pkg/sql" imported and not used
    lint_test.go:1898:
        pkg/cli/democluster/demo_cluster_test.go:312:2: undefined: sqlclustersettings (compile)
```

**Review by rafiss — APPROVED** [review]
> Reviewed 5 of 5 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/116585)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Xiang-Gu)

**Comment by Xiang-Gu** [comment]
TFTR!

bors r+

**Pull request merged**